common/src/View/LaunchGameEngineDialog.cpp

```diff
@@ -218,7 +219,12 @@ namespace TrenchBroom {
                     .arg(IO::pathAsQString(executablePath))
                     .arg(QString::fromStdString(parameters));
 
-                if (!QProcess::startDetached(commandAndArgs)) {
+                const QString oldWorkDir = QDir::currentPath();
+                QDir::setCurrent(workDir);
```

Comment:
Doesn't `QProcess::setWorkingDirectory` work? I feel like that would be more appropriate than changing the working directory globally.

Reply:
The only option is this overload:
https://doc.qt.io/qt-5/qprocess.html#startDetached-1

But to use that, we need to parse the string the user provides into an array, which is complicated because it involves coming up with our own quoting rules. e.g. `quake.exe -basedir "c:\test folder\" -game test`  would get parsed into 5 strings: `quake.exe`, `-basedir`, `c:\test folder\`, `-game`, `test`.

I didn't want to deal with that now as the current method we're doing is already tested on all platforms.

Reply:
Yeah, in that case I agree.